SofaKernel/modules/SofaBaseTopology/RegularGridTopology.cpp

```diff
@@ -171,6 +171,24 @@ Vector3 RegularGridTopology::getPointInGrid(int i, int j, int k) const
     return d_p0.getValue()+dx*i+dy*j+dz*k;
 }
 
+int RegularGridTopology::findPoint(const Vector3& pos)
+{
+    Vector3 p = pos-d_p0.getValue();
+    SReal x = p*dx*inv_dx2;
+    SReal y = p*dy*inv_dy2;
+    SReal z = p*dz*inv_dz2;
+    int ix = int(x+1000000)-1000000; // Do not round toward 0...
+    int iy = int(y+1000000)-1000000;
+    int iz = int(z+1000000)-1000000;
+
+    if (ix < 0 || iy < 0 || iz < 0)
+        return -1;
+
+    if (ix > d_n.getValue()[0] || iy > d_n.getValue()[1]|| iz > d_n.getValue()[2])
```

Comment:
I recommend not writing code that fetch the same data field multiple times in a timestep. 
Two reasons:
- getValue may be a costly, non-constant, operation that involved checking the internal data counters, if a change is detected this trigger a full recomputation of the predecessors in the data graph. 
- the values returned by the different call to getValues() can be different (in a multi-threaded execution)

Reply:
This was a bad copy-paste from another place in Sofa, which I now realise by reading it, isn't even accurate as it will produce wrong indices from the "floor" operation in some cases. I'll rewrite it correctly.